ros/python/py_pubsub/py_pubsub/spatial_mapper.py

```diff
@@ -0,0 +1,311 @@
+import queue
+import socket
+import struct
+import sys
+import time
+import threading
+import pickle
+
+import numpy as np
+import rclpy
+from rclpy.node import Node
+from angel_msgs.msg import ObjectDetection2dSet, SpatialMesh, HeadsetPoseData
+from angel_utils.conversion import to_confidence_matrix
+
+import trimesh
+import trimesh.viewer
+
+
+class SpatialMapSubscriber(Node):
+
+    def __init__(self):
+        super().__init__(self.__class__.__name__)
+
+        self.declare_parameter("spatial_map_topic", "SpatialMapData")
+        self.declare_parameter("det_topic", "ObjectDetections")
+        self.declare_parameter("pose_topic", "HeadsetPoseData")
+
+        self._spatial_map_topic = self.get_parameter("spatial_map_topic").get_parameter_value().string_value
+        self._det_topic = self.get_parameter("det_topic").get_parameter_value().string_value
+        self._pose_topic = self.get_parameter("pose_topic").get_parameter_value().string_value
```

Comment:
I learned that the `declare_parameter` method also returns the parameter instance, so we can collapse these two bits into one.

```suggestion
        self._spatial_map_topic = self.declare_parameter("spatial_map_topic", "SpatialMapData").get_parameter_value().string_value
        self._det_topic = self.declare_parameter("det_topic", "ObjectDetections").get_parameter_value().string_value
        self._pose_topic = self.declare_parameter("pose_topic", "HeadsetPoseData").get_parameter_value().string_value
```